core/src/datayoga_core/expression.py

```diff
@@ -55,35 +59,48 @@ def search(self, data: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
         """
         pass
 
+    @abstractmethod
+    def search_bulk(self, data: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+        """Executes the expression in bulk on a list of data entries
 
-class SQLExpression(Expression):
-    def compile(self, expression: str):
-        # we turn off check_same_thread to gain performance benefit by reusing the same connection object. safe to use since we are only creating in memory structures
-        self.conn = sqlite3.connect(":memory:", check_same_thread=False)
-        self.expression = expression
+        Args:
+            data (List[Dict[str, Any]]): Data
+
+        Returns:
+            List[Dict[str, Any]]: Transformed data
+        """
+        pass
 
-    def filter(self, data: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    def filter(self, data: List[Dict[str, Any]],tombstone:bool = False) -> List[Union[Dict[str, Any],None]]:
```

Comment:
add space after comma (I wonder why it doesn't format it for you this way automatically).